exporter/loggingexporter/logging_exporter.go

```diff
@@ -25,18 +28,128 @@ import (
 	"github.com/open-telemetry/opentelemetry-collector/exporter/exporterhelper"
 )
 
+type traceDataBuffer struct {
+	str strings.Builder
+}
+
+func (b *traceDataBuffer) logEntry(format string, a ...interface{}) {
+	b.str.WriteString(fmt.Sprintf(format, a...))
+	b.str.WriteString("\n")
+}
+
+func (b *traceDataBuffer) logAttr(label string, value string) {
+	b.logEntry("    %-12s: %s", label, value)
+}
+
+func (b *traceDataBuffer) logMap(label string, data *map[string]string) {
```

Comment:
Unclear why the map is passed by pointer here. Is it a performance optimization? Passing map by pointer for performance reasons is unnecessary (unless you need to modify the variable at callsite, which is not the case here) because `map` type already has reference semantics in Go and designed to be passed by value.

It is also not needed if you want to protect from nil maps. It is fine to get keys from a zero initialized (nil) map. From https://golang.org/ref/spec#Map_types
>A nil map is equivalent to an empty map except that no elements may be added.

Reply:
Very good point, thank you!